### PR TITLE
Add charts to dashboard

### PR DESCRIPTION
**Pull request by ilayda21**
## Done
- "Age of pending applications" and "Performance overview" charts are implemented.
- User's full name is retrieved from an endpoint and displayed.
- Number of pending applications are displayed.

## QA
Setup:
- Run server as Peter: `dotrun --env DEBUG_USER_EMAIL=peter.mahnke@canonical.com`

Validation:
- View charts at http://0.0.0.0:8501/
- Verify full name is displayed. (it may not be correct if you're not login with Peter's credentials. It should display the person's name who logged in.)
- Verify number of applications that are pending.
- Verify the "Age of pending applications" chart is displayed.
- Verify the "Performance overview" chart is displayed

## Issue
Fixes #121 #119 #124

### CODE REVIEW

File: static/js/components/PendingChart/PendingChart.tsx [hunk 1]
```diff
@@ -52,33 +50,34 @@ export const PendingChart = () => {
...
       if (
+        isAllSelected ||
         selectedJobFilters.length === 0 ||
         selectedJobFilters.find((filter) => filter === jobApplication.job) !==
           undefined
       ) {
         Object.entries(jobApplication.pending_applications).forEach((entry) => {
           currentPendingDay = +entry[0].split("_")[0];
-
+          pendingCount = entry[1] as number;
```
Comment (hatched) [inline static/js/components/PendingChart/PendingChart.tsx:70]: Whenever you have to use `as` that's usually a code smell that you've incorrectly typed something or that the code is hard to follow. In this case it appears to be lacking sufficient typing of the data types in `rawData`.

Also you're relying on the closure for the value in `rawData`. In a utility function like this you should be able to run it independently. You can pass in all of the necessary data and then return the processed data, setting it outside of the utility function.

Let me know if you have any questions about this.

File: static/js/components/PendingChart/PendingChart.tsx [hunk 2]
```diff
@@ -52,33 +50,34 @@ export const PendingChart = () => {
...
+    let pendingCount: number;
     rawData.forEach((jobApplication) => {
       if (
+        isAllSelected ||
         selectedJobFilters.length === 0 ||
         selectedJobFilters.find((filter) => filter === jobApplication.job) !==
           undefined
       ) {
         Object.entries(jobApplication.pending_applications).forEach((entry) => {
           currentPendingDay = +entry[0].split("_")[0];
```
Comment (hatched) [inline static/js/components/PendingChart/PendingChart.tsx:69]: Because you're dealing with data from a third party API this is pretty fragile:

  - `entry[0]` must be a string or you'll get a syntax error.
  - the `0` index must be an integer else you'll end up with `NaN`.

You might find a regex to be more resilient to bad data.... something like: 

```regex
(^\d+)
```
(untested)

Comment (ilayda21) [reply static/js/components/PendingChart/PendingChart.tsx]: I deleted that part.

File: static/js/components/PendingChart/PendingChart.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    let totalPendingCount = 0;
+    dataToFilter.forEach((jobApplication) => {
+      if (
+        filter.length === 0 ||
+        filter.findIndex(
+          (application) => application.value === jobApplication.job
+        ) !== -1
+      ) {
+        Object.keys(ReviewInterval).forEach((interval) => {
+          if (isNaN(Number(interval))) {
```
Comment (mrgnr) [inline static/js/components/PendingChart/PendingChart.tsx:88]: Is this check necessary? It will always return true given how ReviewInterval is defined.

Comment (ilayda21) [reply static/js/components/PendingChart/PendingChart.tsx]: i changed ReviewInterval's type enum to array so this check is removed.
